operator/src/utils/mod.rs

```diff
@@ -318,3 +318,21 @@ pub fn generate_random_secret(
     rng.fill_bytes(&mut secret_bytes);
     hex::encode(secret_bytes)
 }
+
+/// Apply override env vars to an existing env var list
+pub fn override_env_vars(env: &mut Vec<EnvVar>, overrides: Option<BTreeMap<String, String>>) {
```

Comment:
think this could be `&Option<BTreeMap<String, String>>` since you take a reference to iterate and it would remove the clone.. it isn't important for perf, but I like to make the api match the requirement (i.e make the caller clone if I needed owned, or take a ref if we don't.. but taking a ref and then cloning is usually a bad sign).